pymess/models/emails.py

```diff
@@ -24,6 +28,18 @@
 )
 
 
+def generate_filename(instance, filename):
+    return Path(settings.EMAIL_STORAGE_PATH) / 'contents/content_{}.txt'.format(uuid4())
```

Comment:
There are basically 3 ways how to deal with the filename:

- Use static name and let Django storage engine append its random 7-character string to it. I think this is not a good idea, since generating such strings, especially when amount of e-mails is high, would be slow. Not mentioning that 7-character string has quite limited space of values.
- Use PK (currently integer). In this case, if someone forgets to make the storage private, it might be easy to guess name of existing files and read them.
- Use UUID. Is quick, has pretty large space of values and therefore is hard to guess.

I've chosen option "3".

Reply:
Maybe the messages shoud be stored into subdirs by template name?

Reply:
> Maybe the messages shoud be stored into subdirs by template name?
I think that's not necessary (I don't see any real benefit of that). Moreover, template slug is not mandatory.